packages/frontend/core/src/modules/workbench/browser-adapter.ts

```diff
@@ -0,0 +1,88 @@
+import { useLiveData } from '@toeverything/infra/livedata';
+import { type Location } from 'history';
+import { useEffect } from 'react';
+// eslint-disable-next-line @typescript-eslint/no-restricted-imports
+import { useLocation, useNavigate } from 'react-router-dom';
+
+import type { Workbench } from './workbench';
+
+export function useBindWorkbenchToBrowserRouter(
+  workbench: Workbench,
+  basename: string
+) {
+  const navigate = useNavigate();
+  const browserLocation = useLocation();
+
+  const view = useLiveData(workbench.activeView);
+
+  useEffect(() => {
+    return view.history.listen(update => {
+      if (update.action === 'POP' || update.action === 'REPLACE') {
+        throw new Error(
+          'POP and REPLACE view history is not allowed on browser'
+        );
+      }
```

Comment:
Mind explain why replace is not supported? It is useful for cases like redirect

Reply:
replace is now supported !